### DISCUSSION TITLE
[v4] animation-range and animation-timeline

### DISCUSSION THREAD

**Discussion by bobbymannino** [tailwindlabs/tailwindcss#15086] [Ideas]
In v4 there is a lot of syntax for future css and it is awesome i love it, but what is missing is css scroll based animations, you can use your own `animate-fade-in-up: fade-in-up linear both` but you cannot set animation range or timeline, or i am just missing it, this would be a great feature to have in v4 before its initial release.

Syntax could be something like `timeline-view-block` `range-from-entry-0%` `range-to-entry-100%`?

Edit: im aware you can add it as a utility but i mean baked in

**Comment by stevenvachon** [comment]
For now there's https://npmjs.com/@adam.plesnik/tailwindcss-scroll-driven-animations
